converter/diag/diagnostics.go

```diff
@@ -81,3 +81,14 @@ func (ds Diagnostics) Error() string {
 
 	return sb.String()
 }
+
+func (ds Diagnostics) GenerateReport(reportType string, filename string) error {
+	switch reportType {
+	case Text:
+		return generateTextReport(ds, filename)
+	case HTML:
+		return generateHTMLReport(ds, filename)
+	default:
+		return fmt.Errorf("unsupported diagnostic report type %q", reportType)
+	}
```

Comment:
Idiomatically, this should accept a `io.Writer` argument instead of always writing the report to a file.